proc_test.go

```diff
@@ -14,14 +14,49 @@ import (
 	"path/filepath"
 	"testing"
 
+	"gioui.org/app"
 	"gioui.org/app/headless"
+	"gioui.org/io/event"
 	"gioui.org/io/system"
 	"gioui.org/op"
 	"github.com/go-p5/p5/internal/cmpimg"
 )
 
 var GenerateTestData = flag.Bool("regen", false, "Uses the current state to regenerate the test data.")
 
+// testAppWindowFunc returns a function that creates Test App.Window
+var testAppWindowFunc = func(opts ...app.Option) appWindow {
+	return newTestGioWindow(opts...)
+}
+
+// testAppWindow implements the appWindow interface and its purpose is to be used
+// in tests
+type testAppWindow struct {
+	out     chan event.Event
+	options []app.Option
+}
+
+func newTestGioWindow(opts ...app.Option) *testAppWindow {
+	return &testAppWindow{
+		out:     make(chan event.Event),
+		options: opts,
+	}
+}
+
+func (t *testAppWindow) Events() <-chan event.Event {
+	return t.out
+}
+
+func (t *testAppWindow) Invalidate() {
+	t.out <- system.FrameEvent{
+		Frame: func(ops *op.Ops) {},
+	}
+}
+
+func (t *testAppWindow) Close() {
+	t.out <- system.DestroyEvent{}
+}
```

Comment:
I don't think there's a need to implement the interface as a pointer-receiver, is there?